src/main/java/it/smartcommunitylab/aac/files/persistence/FileInfoRepository.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.smartcommunitylab.aac.files.persistence;
+
+import java.util.List;
+
+import org.springframework.stereotype.Repository;
+
+import it.smartcommunitylab.aac.repository.CustomJpaRepository;
+
+@Repository
+public interface FileInfoRepository extends CustomJpaRepository<FileInfo, String> {
+
+	FileInfo findByRealmAndId(String realm, String id);
```

Comment:
Id is unique and there is already the default finder. This should be `findByRealmAndName`, where `name` should be unique per realm.
Note: name could contain a path here.